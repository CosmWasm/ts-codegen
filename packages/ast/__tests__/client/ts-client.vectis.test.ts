import { ExecuteMsg } from '@cosmwasm/ts-codegen-types';

import {
  createExecuteClass,
  createExecuteInterface,
  createQueryClass,
  createTypeInterface
} from '../../src';
import {
  expectCode,
  getLegacyFixture,
  makeContext,
} from '../../test-utils';

const vectisCanExecuteRelayResponse = getLegacyFixture('vectis/govec', '/can_execute_relay_response.json');
const vectisCosmosMsgForEmpty = getLegacyFixture('vectis/govec', '/cosmos_msg_for__empty.json');
const vectisExecuteMsgForEmpty = getLegacyFixture('vectis/govec', '/execute_msg_for__empty.json');
const vectisInfoResponse = getLegacyFixture('vectis/govec', '/info_response.json');
const vectisRelayTransaction = getLegacyFixture('vectis/govec', '/relay_transaction.json');

it('cosmos_msg_for__empty', () => {
  const ctx = makeContext(vectisCosmosMsgForEmpty);
  expectCode(createTypeInterface(ctx, vectisCosmosMsgForEmpty));
});

it('execute_msg_for__empty', () => {
  const ctx = makeContext(vectisExecuteMsgForEmpty);
  expectCode(createTypeInterface(ctx, vectisExecuteMsgForEmpty));
});

it('can_execute_relay_response', () => {
  const ctx = makeContext(vectisCanExecuteRelayResponse);
  expectCode(createTypeInterface(ctx, vectisCanExecuteRelayResponse));
});

it('info_response', () => {
  const ctx = makeContext(vectisInfoResponse);
  expectCode(createTypeInterface(ctx, vectisInfoResponse));
});

it('relay_transaction', () => {
  const ctx = makeContext(vectisRelayTransaction);
  expectCode(createTypeInterface(ctx, vectisRelayTransaction));
});

it('query classes', () => {
  const ctx = makeContext(vectisCosmosMsgForEmpty);
  expectCode(
    createQueryClass(
      ctx,
      'SG721QueryClient',
      'SG721ReadOnlyInstance',
      // @ts-ignore
      vectisCosmosMsgForEmpty
    )
  );
});

it('query classes', () => {
  const ctx = makeContext(vectisExecuteMsgForEmpty);
  expectCode(
    createQueryClass(
      ctx,
      'SG721QueryClient',
      'SG721ReadOnlyInstance',
      // @ts-ignore
      vectisExecuteMsgForEmpty
    )
  );
});

it('execute classes array types', () => {
  const ctx = makeContext(vectisExecuteMsgForEmpty);
  expectCode(
    createExecuteClass(
      ctx,
      'SG721Client',
      'SG721Instance',
      null,
      // @ts-ignore
      vectisExecuteMsgForEmpty as ExecuteMsg
    )
  );
});

it('execute interfaces no extends', () => {
  const ctx = makeContext(vectisExecuteMsgForEmpty);
  expectCode(
    // @ts-ignore
    createExecuteInterface(ctx, 'SG721Instance', null, vectisExecuteMsgForEmpty)
  );
});
