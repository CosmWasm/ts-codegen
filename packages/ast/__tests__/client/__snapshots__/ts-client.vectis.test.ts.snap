// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can_execute_relay_response 1`] = `
"export interface CanExecuteRelayResponse {
  canExecute: boolean;
}"
`;

exports[`cosmos_msg_for__empty 1`] = `"export type CosmosMsg_for_Empty = CosmosMsg_for_Empty;"`;

exports[`execute classes array types 1`] = `
"export class SG721Client implements SG721Instance {
  client: SigningCosmWasmClient;
  sender: string;
  contractAddress: string;
  constructor(client: SigningCosmWasmClient, sender: string, contractAddress: string) {
    this.client = client;
    this.sender = sender;
    this.contractAddress = contractAddress;
    this.execute = this.execute.bind(this);
    this.revertFreezeStatus = this.revertFreezeStatus.bind(this);
    this.relay = this.relay.bind(this);
    this.rotateUserKey = this.rotateUserKey.bind(this);
    this.addRelayer = this.addRelayer.bind(this);
    this.removeRelayer = this.removeRelayer.bind(this);
    this.updateGuardians = this.updateGuardians.bind(this);
    this.updateLabel = this.updateLabel.bind(this);
  }
  execute = async ({
    msgs
  }: {
    msgs: CosmosMsg_for_Empty[];
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      execute: {
        msgs
      }
    }, fee_, memo_, funds_);
  };
  revertFreezeStatus = async (fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      revert_freeze_status: {}
    }, fee_, memo_, funds_);
  };
  relay = async ({
    transaction
  }: {
    transaction: RelayTransaction;
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      relay: {
        transaction
      }
    }, fee_, memo_, funds_);
  };
  rotateUserKey = async ({
    newUserAddress
  }: {
    newUserAddress: string;
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      rotate_user_key: {
        new_user_address: newUserAddress
      }
    }, fee_, memo_, funds_);
  };
  addRelayer = async ({
    newRelayerAddress
  }: {
    newRelayerAddress: Addr;
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      add_relayer: {
        new_relayer_address: newRelayerAddress
      }
    }, fee_, memo_, funds_);
  };
  removeRelayer = async ({
    relayerAddress
  }: {
    relayerAddress: Addr;
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      remove_relayer: {
        relayer_address: relayerAddress
      }
    }, fee_, memo_, funds_);
  };
  updateGuardians = async ({
    guardians,
    newMultisigCodeId
  }: {
    guardians: Guardians;
    newMultisigCodeId?: number;
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      update_guardians: {
        guardians,
        new_multisig_code_id: newMultisigCodeId
      }
    }, fee_, memo_, funds_);
  };
  updateLabel = async ({
    newLabel
  }: {
    newLabel: string;
  }, fee_: number | StdFee | "auto" = "auto", memo_?: string, funds_?: Coin[]): Promise<ExecuteResult> => {
    return await this.client.execute(this.sender, this.contractAddress, {
      update_label: {
        new_label: newLabel
      }
    }, fee_, memo_, funds_);
  };
}"
`;

exports[`execute interfaces no extends 1`] = `
"export interface SG721Instance {
  contractAddress: string;
  sender: string;
  execute: ({
    msgs
  }: {
    msgs: CosmosMsg_for_Empty[];
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  revertFreezeStatus: (fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  relay: ({
    transaction
  }: {
    transaction: RelayTransaction;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  rotateUserKey: ({
    newUserAddress
  }: {
    newUserAddress: string;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  addRelayer: ({
    newRelayerAddress
  }: {
    newRelayerAddress: Addr;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  removeRelayer: ({
    relayerAddress
  }: {
    relayerAddress: Addr;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  updateGuardians: ({
    guardians,
    newMultisigCodeId
  }: {
    guardians: Guardians;
    newMultisigCodeId?: number;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
  updateLabel: ({
    newLabel
  }: {
    newLabel: string;
  }, fee_?: number | StdFee | "auto", memo_?: string, funds_?: Coin[]) => Promise<ExecuteResult>;
}"
`;

exports[`execute_msg_for__empty 1`] = `"export type ExecuteMsg_for_Empty = ExecuteMsg_for_Empty;"`;

exports[`info_response 1`] = `
"export interface InfoResponse {
  codeId: number;
  guardians: Addr[];
  isFrozen: boolean;
  label: string;
  multisigAddress?: Addr;
  multisigCodeId: number;
  nonce: number;
  relayers: Addr[];
  userAddr: Addr;
  version: ContractVersion;
}"
`;

exports[`query classes 1`] = `
"export class SG721QueryClient implements SG721ReadOnlyInstance {
  client: CosmWasmClient;
  contractAddress: string;
  constructor(client: CosmWasmClient, contractAddress: string) {
    this.client = client;
    this.contractAddress = contractAddress;
    this.bank = this.bank.bind(this);
    this.custom = this.custom.bind(this);
    this.staking = this.staking.bind(this);
    this.distribution = this.distribution.bind(this);
    this.wasm = this.wasm.bind(this);
  }
  bank = async (bankMsg: BankMsg): Promise<BankResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      bank: bankMsg
    });
  };
  custom = async (): Promise<CustomResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      custom: {}
    });
  };
  staking = async (stakingMsg: StakingMsg): Promise<StakingResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      staking: stakingMsg
    });
  };
  distribution = async (distributionMsg: DistributionMsg): Promise<DistributionResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      distribution: distributionMsg
    });
  };
  wasm = async (wasmMsg: WasmMsg): Promise<WasmResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      wasm: wasmMsg
    });
  };
}"
`;

exports[`query classes 2`] = `
"export class SG721QueryClient implements SG721ReadOnlyInstance {
  client: CosmWasmClient;
  contractAddress: string;
  constructor(client: CosmWasmClient, contractAddress: string) {
    this.client = client;
    this.contractAddress = contractAddress;
    this.execute = this.execute.bind(this);
    this.revertFreezeStatus = this.revertFreezeStatus.bind(this);
    this.relay = this.relay.bind(this);
    this.rotateUserKey = this.rotateUserKey.bind(this);
    this.addRelayer = this.addRelayer.bind(this);
    this.removeRelayer = this.removeRelayer.bind(this);
    this.updateGuardians = this.updateGuardians.bind(this);
    this.updateLabel = this.updateLabel.bind(this);
  }
  execute = async ({
    msgs
  }: {
    msgs: CosmosMsg_for_Empty[];
  }): Promise<ExecuteResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      execute: {
        msgs
      }
    });
  };
  revertFreezeStatus = async (): Promise<RevertFreezeStatusResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      revert_freeze_status: {}
    });
  };
  relay = async ({
    transaction
  }: {
    transaction: RelayTransaction;
  }): Promise<RelayResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      relay: {
        transaction
      }
    });
  };
  rotateUserKey = async ({
    newUserAddress
  }: {
    newUserAddress: string;
  }): Promise<RotateUserKeyResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      rotate_user_key: {
        new_user_address: newUserAddress
      }
    });
  };
  addRelayer = async ({
    newRelayerAddress
  }: {
    newRelayerAddress: Addr;
  }): Promise<AddRelayerResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      add_relayer: {
        new_relayer_address: newRelayerAddress
      }
    });
  };
  removeRelayer = async ({
    relayerAddress
  }: {
    relayerAddress: Addr;
  }): Promise<RemoveRelayerResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      remove_relayer: {
        relayer_address: relayerAddress
      }
    });
  };
  updateGuardians = async ({
    guardians,
    newMultisigCodeId
  }: {
    guardians: Guardians;
    newMultisigCodeId?: number;
  }): Promise<UpdateGuardiansResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      update_guardians: {
        guardians,
        new_multisig_code_id: newMultisigCodeId
      }
    });
  };
  updateLabel = async ({
    newLabel
  }: {
    newLabel: string;
  }): Promise<UpdateLabelResponse> => {
    return this.client.queryContractSmart(this.contractAddress, {
      update_label: {
        new_label: newLabel
      }
    });
  };
}"
`;

exports[`relay_transaction 1`] = `
"export interface RelayTransaction {
  message: Binary;
  nonce: number;
  signature: Binary;
  userPubkey: Binary;
}"
`;
